[CLI] update Parameter Store reference for v11 (#5174)
aws-amplify/docs: src/pages/cli/reference/ssm-parameter-store.mdx

With this change the values historically stored in the TPI file are now being copied to Parameter Store. We are not removing the TPI file, Parameter Store is being used as a backup. If Amplify CLI needs a value that is not present in the TPI file, it will fetch the value from Parameter Store. The values in Parameter Store are updated after a successful Amplify deployment.

Developers can now seamlessly move between various environments like different developer machines and even to AWS Amplify Studio.

> <Callout>Note: When an Amplify environement is deleted, all parameters under that environment name are deleted in Parmater Store. When an Amplify project is deleted, all parameters for that appId are deleted along with it.</Callout>

Reviewer: ```suggestion
<Callout>Note: When an Amplify environement is deleted, all parameters under that environment name are deleted in Paramater Store. When an Amplify project is deleted, all parameters for that `appId` are deleted along with it.</Callout>
```

> <Callout>Note: When an Amplify environement is deleted, all parameters under that environment name are deleted in Paramater Store. When an Amplify project is deleted, all parameters for that `appId` are deleted along with it.</Callout>

Author: ```suggestion
<Callout>Note: When an Amplify environment is deleted, all parameters under that environment name are deleted in Parameter Store. When an Amplify project is deleted, all parameters for that `appId` are deleted along with it.</Callout>
```

## Manually creating parameters

In some multi-environment scenarios, it is helpful to create parameters in Parameter Store manually. To do so, head to the AWS Console and search for SSM.

> ![image](/images/cli/parameter-store/select-ssm-console.png)

Reviewer: ```suggestion
![AWS Console's search with "SSM"](/images/cli/parameter-store/select-ssm-console.png)
```

Once the SSM console has been opened, navigate to Parameter Store.

> ![image](/images/cli/parameter-store/select-parameter-store.png)

Reviewer: ```suggestion
![AWS Systems Manager side navigation with "Parameter Store" highlighted](/images/cli/parameter-store/select-parameter-store.png)
```

With Parameter Store opened, select "Create Parameter".

> ![image](/images/cli/parameter-store/view-parameter-store.png)

Reviewer: ```suggestion
![AWS SSM Parameter Store's landing page with a list of parameters](/images/cli/parameter-store/view-parameter-store.png)
```

Amplify parameter keys must follow a strict naming pattern:

> /amplify/<appId>/<environment-name>/AMPLIFY_<category-name>_<resource-name>_<parameter-name>

Reviewer: ```suggestion
```
/amplify/<appId>/<environment-name>/AMPLIFY_<category-name>_<resource-name>_<parameter-name>
```
```

Only Standard parameters with the default String type are supported.

> ![image](/images/cli/parameter-store/create-parameter.png)

Reviewer: ```suggestion
![AWS SSM Parameter Store's "Create Parameter" form with Amplify's parameter naming pattern](/images/cli/parameter-store/create-parameter.png)
```
